### ISSUE TITLE
__name__ variable is not __main__ as expected

### ISSUE THREAD

**Issue by ctrueden** [scijava/scripting-jython#9]
A very common idiom in Python/Jython is to write:
```python
if __name__ == "__main__":
    main()
```

The Python documentation [states](https://docs.python.org/2/library/__main__.html) that this idiom is supposed to work, and the Jython documentation [agrees](http://www.jython.org/docs/library/__main__.html). And this particular idiom also appears in the Jython source code many times.

If you boot up a Jython 2.7.1 interpreter (e.g. via [`jrun org.python:jython-standalone`](https://github.com/ctrueden/jrun)), and do `print __name__`, it indeed says `__main__`.

However, the way the SciJava script framework invokes the Jython JSR-223 `ScriptEngineFactory` (bundled with Jython since 2.7.x) somehow results in `__name__` being set to `__builtin__` instead.

This discrepancy needs to be addressed so that the standard Python paradigm for handling main methods works as intended from the SciJava Script Editor and elsewhere.

**Comment by jeff5** [comment]
Thanks to Alex Gobbo raising https://bugs.jython.org/issue2846 (fixed) the behaviour of Jython 2.7.2 ought to suit you better in this respect. This should appear in a beta 3 in the next week.

**Comment by ctrueden** [comment]
Thank you @jeff5 for the detailed investigation and fix, and for following up here! Much appreciated.

**Comment by hinerm** [comment]
Seems to be fixed now with 2.72, e.g.:

```java
final String script = "#@output String res\n" + "res = __name__\n";
for (int i =0; i<5; i++) {
	final ScriptModule m = scriptService.run("add" + i + ".py", script, true).get();
	final Object result = m.getInfo().getLanguage().decode(m.getOutput("res"));
	System.out.println(result);
}
```

outputs:
```
__main__
__main__
__main__
__main__
__main__
```
